webvh/webvh/did/tests/test_endorsement_namager.py

```diff
@@ -0,0 +1,126 @@
+import asyncio
+from unittest import IsolatedAsyncioTestCase
+
+from acapy_agent.connections.models.conn_record import ConnRecord
+from acapy_agent.messaging.models.base import BaseModelError
+from acapy_agent.protocols.coordinate_mediation.v1_0.route_manager import RouteManager
+from acapy_agent.protocols.out_of_band.v1_0.manager import OutOfBandManager
+from acapy_agent.tests import mock
+from acapy_agent.utils.testing import create_test_profile
+
+from ..endorsement_manager import EndorsementManager
+from ..exceptions import ConfigurationError, EndorsementError
+
+
+class TestEndorsementManager(IsolatedAsyncioTestCase):
+    async def asyncSetUp(self):
+        self.profile = await create_test_profile({"wallet.type": "askar-anoncreds"})
+
+    @mock.patch.object(EndorsementManager, "_get_active_endorser_connection")
+    async def test_auto_endorsement_setup_as_endorser(
+        self, mock_get_active_endorser_connection
+    ):
+        self.profile.settings.set_value(
+            "plugin_config",
+            {"did-webvh": {"role": "endorser"}},
+        )
+        await EndorsementManager(self.profile).auto_endorsement_setup()
+        assert not mock_get_active_endorser_connection.called
+
+    async def test_auto_endorsement_setup_as_author_no_server_url(self):
+        self.profile.settings.set_value(
+            "plugin_config",
+            {"did-webvh": {"role": "author"}},
```

Comment:
I would set the role as the "controller", as in the "controller" of the did document.
https://www.w3.org/TR/did-core/#did-controller